File: src/arithmetic_multiplication_easy.py
"""
arithmetic - multiplication (easy)
Generated: 2026-02-22T03:38:50.816491
"""

from problem_utils import *

def generate():
    problem_type = randint(1, 4)
    
    if problem_type == 1:
        # Single digit multiplication (1000-1100 ELO)
        a_val = randint(2, 9)
        b_val = randint(2, 9)
        ans = a_val * b_val
        
        return problem(
            question=f"${a_val} \\times {b_val} = ?$",
            answer=ans,
            difficulty=(1000, 1100),
            topic="arithmetic/multiplication",
            solution=f"${a_val} \\times {b_val} = {ans}$"
        )
    
    elif problem_type == 2:
        # Single digit times double digit (1100-1200 ELO)
        a_val = randint(2, 9)
        b_val = randint(10, 20)
        ans = a_val * b_val
        
        return problem(
            question=f"${a_val} \\times {b_val} = ?$",
            answer=ans,
            difficulty=(1100, 1200),
            topic="arithmetic/multiplication",
            solution=steps(
                f"Multiply ${a_val} \\times {b_val}$",
                f"${a_val} \\times {b_val} = {ans}$"
            )
        )
    
    elif problem_type == 3:
        # Two-digit times single digit with regrouping (1150-1250 ELO)
        tens = randint(2, 9)
        ones = randint(2, 9)
        a_val = tens * 10 + ones
        b_val = randint(2, 9)
        ans = a_val * b_val
        
        return problem(
            question=f"${a_val} \\times {b_val} = ?$",
            answer=ans,
            difficulty=(1150, 1250),
            topic="arithmetic/multiplication",
            solution=steps(
                f"Break down: ${a_val} = {tens * 10} + {ones}$",
                f"${b_val} \\times {tens * 10} = {b_val * tens * 10}$",
                f"${b_val} \\times {ones} = {b_val * ones}$",
                f"${b_val * tens * 10} + {b_val * ones} = {ans}$"
            )
        )
    
    else:
        # Multiply by 10, 100 (1050-1150 ELO)
        a_val = randint(2, 99)
        multiplier = choice([10, 100])
        ans = a_val * multiplier
        
        zeros = "0" if multiplier == 10 else "00"
        
        return problem(
            question=f"${a_val} \\times {multiplier} = ?$",
            answer=ans,
            difficulty=(1050, 1150),
            topic="arithmetic/multiplication",
            solution=steps(
                f"When multiplying by ${multiplier}$, add {len(zeros)} zero(s) to the end",
                f"${a_val} \\times {multiplier} = {ans}$"
            )
        )

emit(generate())

File: src/arithmetic_multiplication_easy.jl
# arithmetic - multiplication (easy)
# Generated: 2026-03-08T19:46:27.786967

include(joinpath(@__DIR__, "..", "julia", "src", "ProblemUtils.jl"))
using .ProblemUtils

@script begin
    set_topic!("arithmetic/multiplication")
    
    problem_type = choice(1:5)
    
    if problem_type == 1
        # Two-digit by one-digit multiplication (700-900 ELO)
        a = randint(10, 99)
        b = randint(2, 9)
        ans = a * b
        problem(
            question="Calculate: \$$(a) \\times $(b)\$",
            answer=ans,
            difficulty=(700, 900),
            solution=steps(
                sol("Problem", "$(a) × $(b)"),
                "Break down: $(a) = $(div(a, 10) * 10) + $(a % 10)",
                "$(div(a, 10) * 10) × $(b) = $(div(a, 10) * 10 * b), and $(a % 10) × $(b) = $((a % 10) * b)",
                sol("Answer", ans)
            ),
            time=60
        )
    elseif problem_type == 2
        # Two-digit by two-digit multiplication (800-1000 ELO)
        a = randint(11, 49)
        b = randint(11, 49)
        ans = a * b
        problem(
            question="Calculate: \$$(a) \\times $(b)\$",
            answer=ans,
            difficulty=(800, 1000),
            solution=steps(
                sol("Problem", "$(a) × $(b)"),
                "Use standard multiplication algorithm",
                "$(a) × $(b % 10) = $(a * (b % 10))",
                "$(a) × $(div(b, 10) * 10) = $(a * div(b, 10) * 10)",
                sol("Answer", ans)
            ),
            time=90
        )
    elseif problem_type == 3
        # Three-digit by one-digit multiplication (750-950 ELO)
        a = randint(100, 999)
        b = randint(2, 9)
        ans = a * b
        problem(
            question="Calculate: \$$(a) \\times $(b)\$",
            answer=ans,
            difficulty=(750, 950),
            solution=steps(
                sol("Problem", "$(a) × $(b)"),
                "Multiply each digit by $(b) from right to left",
                "Apply carrying as needed",
                sol("Answer", ans)
            ),
            time=90
        )
    elseif problem_type == 4
        # Multiplication with a multiple of 10 (700-850 ELO)
        a = randint(12, 99)
        b = randint(2, 9) * 10
        ans = a * b
        problem(
            question="Calculate: \$$(a) \\times $(b)\$",
            answer=ans,
            difficulty=(700, 850),
            solution=steps(
                sol("Problem", "$(a) × $(b)"),
                "Notice that $(b) = $(div(b, 10)) × 10",
                "First: $(a) × $(div(b, 10)) = $(a * div(b, 10))",
                "Then multiply by 10: $(a * div(b, 10)) × 10 = $(ans)",
                sol("Answer", ans)
            ),
            time=75
        )
    else
        # Word problem with multiplication (900-1200 ELO)
        items_per_box = randint(12, 48)
        num_boxes = randint(5, 25)
        ans = items_per_box * num_boxes
        
        item_names = ["pencils", "markers", "erasers", "notebooks", "crayons", "pens"]
        item = choice(item_names)
        
        problem(
            question="A store has $(num_boxes) boxes of $(item). Each box contains $(items_per_box) $(item). How many $(item) does the store have in total?",
            answer=ans,
            difficulty=(900, 1200),
            solution=steps(
                "Identify the multiplication needed: number of boxes × items per box",
                sol("Calculation", "$(num_boxes) × $(items_per_box)"),
                "Perform the multiplication",
                sol("Answer", "$(ans) $(item)")
            ),
            time=120
        )
    end
end